packages/platforms/react-native/lib/id-generator.ts

```diff
@@ -11,18 +16,55 @@ function toHex (value: number): string {
   return hex
 }
 
-const idGenerator: IdGenerator = {
-  generate (bits: BitLength): string {
-    const bytes = bits / 8
+export function createRandomString (): string {
+  let random = ''
 
-    let randomValue = ''
+  for (let i = 0; i < POOL_SIZE; i++) {
+    random += toHex((Math.random() * 255) | 0)
+  }
 
-    for (let i = 0; i < bytes; i++) {
-      randomValue += toHex((Math.random() * 255) | 0)
-    }
+  return random
+}
+
+function createIdGenerator (NativeBugsnagPerformance: NativeBugsnag | null): IdGenerator {
+  // If the native module is not available for any reason, fall back to a JS implementation
+  const requestEntropy = NativeBugsnagPerformance ? NativeBugsnagPerformance.requestEntropy : createRandomString
+  const requestEntropyAsync = NativeBugsnagPerformance ? NativeBugsnagPerformance.requestEntropyAsync : async () => createRandomString()
 
-    return randomValue
+  // initialise the pool synchronously
+  const randomValues = requestEntropy()
+  let randomPool = randomValues.length > 0 ? randomValues : createRandomString()
+
+  const regeneratePool = async () => {
+    const randomValues = await requestEntropyAsync()
+    randomPool = randomValues.length > 0 ? randomValues : createRandomString()
   }
+
+  let numCalls = 0
+  const idGenerator: IdGenerator = {
+    generate (bits: BitLength): string {
+      const chars = bits / 4
+      let id = ''
+
+      // pick characters from the pool semi-randomly
+      for (let i = 0; i < chars; i++) {
+        const randomIdx = (Math.random() * (randomPool.length - 1)) | 0
+        id += randomPool[randomIdx]
+      }
+
+      numCalls++
+
+      // if the max number of calls has been reached, refresh the pool asynchronously
+      if (numCalls >= MAX_CALLS) {
```

Comment:
Could do a little bit of squishening if you don't hate it:

```suggestion
      // if the max number of calls has been reached, refresh the pool asynchronously
      if (++numCalls >= MAX_CALLS) {
```

Reply:
Always in favour of a bit of squishening